File: E4/e04_1/app/src/main/java/com/example/e04_1/ContentActivity.java
package com.example.e04_1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class ContentActivity extends AppCompatActivity {

    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Initialize Firebase Auth
        mAuth = FirebaseAuth.getInstance();

        TextView mainText = findViewById(R.id.main_text);
        FirebaseUser user = mAuth.getCurrentUser();

        if (user != null) {
            mainText.setText(user.getEmail());
        }
    }
}

File: 2022-SOPM/E04/app/src/main/java/com/example/e04_00/ContentActivity.kt
package com.example.e04_00

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.e04_00.utilities.PASSWORD_KEY
import com.example.e04_00.utilities.USERNAME_KEY

class ContentActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_content)

        val mainActivityIntent = intent

        val username = mainActivityIntent.getStringExtra(USERNAME_KEY)
        val password = mainActivityIntent.getStringExtra(PASSWORD_KEY)

        val welcomeMessage: TextView = findViewById(R.id.welcome_message)

        welcomeMessage.text = "Hello $username, your password is $password"
    }
}